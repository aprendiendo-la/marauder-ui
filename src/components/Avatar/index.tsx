import * as React from "react";
import styled from "styled-components";

interface ContainerProps {
  readonly src: string;
  readonly width: number;
  readonly height: number;
  readonly round: boolean;
  readonly borderStyle: string;
}

const Container = styled.div<ContainerProps>`
  margin: 0 auto;
  background-image: url('${props => props.src}');
  background-repeat: no-repeat; 
  background-size: cover;
  width: ${props => props.width}px;
  height: ${props => props.height}px;
  border-radius: ${props => (props.round ? "50%" : "0%")};
  border: ${props => props.borderStyle || "none"};
`;

const getSize = (size: string) => {
  if (size === "small") {
    return 45;
  }
  if (size === "medium") {
    return 75;
  }
  if (size === "large") {
    return 105;
  }
};

interface Props {
  round?: boolean;
  src?: string;
  size?: string;
  borderStyle?: string;
}

const Avatar: React.SFC<Props> = ({ round, src, size, borderStyle }) => {
  return (
    <Container
      src={src}
      width={getSize(size)}
      height={getSize(size)}
      round={round}
      borderStyle={borderStyle}
    />
  );
};

Avatar.defaultProps = {
  src:
    "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
  size: "medium",
  round: false,
  borderStyle: "none"
};

export default Avatar;
