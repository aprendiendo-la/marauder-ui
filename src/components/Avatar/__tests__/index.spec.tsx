import * as React from "react";
import { shallow } from "enzyme";

import Avatar from "../index";

const setUp = (props = {}) => {
    const component = shallow(<Avatar {...props}/>)
    return component;
}

describe('Avatar Component', () => {
    let component : any;

    beforeEach(() => {
        component = setUp();
    })

    it("Renders Avatar component", () => {
        expect(component).toMatchSnapshot()
    });
})
