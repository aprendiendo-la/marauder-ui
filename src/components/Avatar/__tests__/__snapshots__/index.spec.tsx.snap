// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Renders Avatar component 1`] = `
<styled.div
  borderStyle="none"
  height={75}
  round={false}
  src="https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
  width={75}
/>
`;

exports[`Avatar Component renders the avatar 1`] = `
<styled.div
  borderStyle="none"
  height={75}
  round={false}
  src="https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
  width={75}
/>
`;
